terraform/modules/kubernetes/monitoring.tf

```diff
@@ -115,3 +115,33 @@ resource "azurerm_monitor_metric_alert" "redis_memory" {
     ]
   }
 }
+data "azurerm_monitor_diagnostic_categories" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  resource_id = azurerm_postgresql_flexible_server.postgres-server[0].id
+}
+
+resource "azurerm_log_analytics_workspace" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+  name                = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-logs"
+  location            = data.azurerm_resource_group.app-resource-group.location
+  resource_group_name = data.azurerm_resource_group.app-resource-group.name
+  sku                 = "PerGB2018"
+}
+resource "azurerm_monitor_diagnostic_setting" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+  name                       = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-diag"
+  target_resource_id         = azurerm_postgresql_flexible_server.postgres-server[0].id
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.pg_loganalytics[0].id
+  dynamic "log" {
```

Comment:
Please add blank line

Reply:
It is but for some reason when I change it it does not like it could be the change introduced recently, our current version is 3.24.0 but the deprecation is set to kick in in version 4 so safe to leave it in there for now untill we decide to upgrade to a higer version

Reply:
We already used it there: https://github.com/DFE-Digital/teacher-services-cloud/blob/main/cluster/terraform_aks_cluster/analytics.tf#L19
So why can't we use it here?